scipy/_lib/_testutils.py

```diff
@@ -255,3 +257,14 @@ def _get_mem_available():
             return info['memfree'] + info['cached']
 
     return None
+
+
+def _assert_matching_namespace(actual, expected):
```

Comment:
Thinking about this more, I do think we'd still want to assert that the return array of fftfreq is the same array as in `xp`. What I was trying to get at is the `_assert_matching_namespace` function won't work for the array creation functions that need to take in `xp` and `device`. In that case, we'll need something more like `_assert_matching_xp_and_device(actual, expected_xp, expected_device)` - but with a better name than that!

Reply:
Yes I agree, that's what I had in mind. It shouldn't be difficult to implement this since array-api-compat has a `check_device` helper. I can get this working for now, I'm just hesitant to be too set on this since `xp=np, device=None` is new and a different design could be chosen by the SciPy maintainers.

Reply:
Moved to `_array_api.py` and changed to `xp=None`.